radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java

```diff
@@ -78,6 +86,9 @@ private RadixNetworkController(RadixNetwork network, List<RadixNetworkEpic> epic
 			);
 			*/
 
+			// TODO: Move this into a proper reducer framework
+			reducers.forEach(r -> r.accept(action));
+
 			// TODO: also add equals check
```

Comment:
Appreciate this is old - as I understand, this might cause unnecessary updates by calling networkState.onNext when nextState.equals(curState) but nextState != curState (i.e. equals but not same instance). Is this documented anywhere / do we have n issue? Is it a quick fix (looks like it)?

Reply:
Not sure if it's actually correct to check for `equals` here because epics might be dependent on states being emitted if the reducer processes an action even if the two states are equal. Would have to look into our epics and also what the standard is for this. Either way, this is probably a separate task which I'll add to the backlog.